backend/substrapp/tasks/utils.py

```diff
@@ -182,6 +182,8 @@ def container_format_log(container_name, container_logs):
 
 
 def list_files(startpath):
+    if not settings.TASK['LIST_WORKSPACE']:
+        return
```

Comment:
We should never return nothing.
Why not simply use `if settings.TASK['LIST_WORKSPACE']:`